include/mim_control/impedance_controller.hpp

```diff
@@ -19,199 +19,243 @@ namespace mim_control
-                              const pinocchio::Force& feed_forward_force);
-
-    /**
-     * @brief Get the computed torques from the impedance controller.
-     *
-     * @return Eigen::VectorXd&
-     */
-    const Eigen::VectorXd& get_torques();
-
-    /**
-     * @brief Get the computed joint torques from the impedance controller.
-     *
-     * @return Eigen::VectorXd&
-     */
-    const Eigen::VectorXd& get_joint_torques();
-
-    /**
-     * @brief Get the impedance force \f$ f_i \f$ with \f$ \tau = J^T f_i \f$.
-     *
-     * @return Vector6d&
-     */
-    const Vector6d& get_impedance_force();
-
-    /**
-     * @brief Returns the index of the endeffector frame.
-     *
-     * @return pinocchio::FrameIndex
-     */
-    const pinocchio::FrameIndex& get_endframe_index();
-
-private:  // attributes
-    /** @brief Rigid body dynamics model. */
-    pinocchio::Model pinocchio_model_;
-
-    /** @brief Cache of the rigid body dynamics algorithms. */
-    pinocchio::Data pinocchio_data_;
-
-    /** @brief (urdf) name of the root frame. The impedance controller is
-     * computed with respect to this frame. */
-    std::string root_frame_name_;
-
-    /** @brief Index of the root frame in the pinocchio model. */
-    pinocchio::FrameIndex root_frame_index_;
-
-    /** @brief Measured root frame placement. */
-    pinocchio::SE3 root_placement_;
-
-    /** @brief Measured root frame orientation. */
-    pinocchio::SE3 root_orientation_;
-
-    /** @brief Measured root frame velocity. */
-    pinocchio::Motion root_velocity_;
-
-    /** @brief (urdf) name of the end frame. This is the controlled frame. */
-    std::string end_frame_name_;
-
-    /** @brief Index of the end frame in the pinocchio model. */
-    pinocchio::FrameIndex end_frame_index_;
-
-    /** @brief Jacobian of the end frame. */
-    pinocchio::Data::Matrix6x end_jacobian_;
-
-    /** @brief Measured end frame placement. */
-    pinocchio::SE3 end_placement_;
-
-    /** @brief Measured end frame orientation. */
-    pinocchio::SE3 end_orientation_;
-
-    /** @brief Measured end frame velocity. */
-    pinocchio::Motion end_velocity_;
-
-    /** @brief Impedance force. Accessible for machine learning purposes. */
-    Vector6d impedance_force_;
-
-    /** @brief Measured end frame placement in root frame. */
-    pinocchio::SE3 actual_end_frame_placement_;
-
-    /** @brief Measured end frame placement in root frame. */
-    pinocchio::Motion actual_end_frame_velocity_;
-
-    /** @brief SE3 placement error. */
-    Eigen::Matrix<double, 6, 1> err_se3_;
-
-    /** @brief SE3 velocity error. */
-    pinocchio::Motion err_vel_;
-
-    /** @brief Jacobian used in the computation of the impedance. */
-    pinocchio::Data::Matrix6x impedance_jacobian_;
-
-    /** @brief Output torques. */
-    Eigen::VectorXd torques_;
-
-    /** @brief Output torques. */
-    Eigen::VectorXd joint_torques_;
+    class ImpedanceController
```

Comment:
Use [mpi_cmake_modules](https://github.com/machines-in-motion/mpi_cmake_modules) to change the format of your code and make it similar to others files in this repo.

Reply:
It's not easy to review this package because of using a different format.

Reply:
Are you referring to this: https://github.com/machines-in-motion/mpi_cmake_modules/blob/master/doc/clang_format.md? If so, I couldn't get it to work (command not found even though I sourced my workspace folder). For now, I just un-indented the code.

Reply:
try with dash instead of underscore
[run-clang-format](https://github.com/machines-in-motion/mpi_cmake_modules/blob/master/scripts/run-clang-format)

Reply:
After running `run-clang-format absolute-path-to-file absolute-path-to-folder`, I get this error message: `run-clang-format: error: Command 'clang-format --version' failed to start: [Errno 2] No such file or directory: 'clang-format'`. Do you know what is causing it?

Reply:
`sudo apt install clang-format`?

Reply:
Ah yes, thank you